drivers/peci/peci_ite_it8xxx2.c

```diff
@@ -0,0 +1,355 @@
+	.alts_list = peci_alt_inst,
+};
+
+static struct peci_it8xxx2_data peci_it8xxx2_data0;
+
+/* ITE IT8XXX2 PECI Functions */
+
+static void peci_it8xxx2_init_vtts(struct peci_it8xxx2_regs *reg_base,
+					enum peci_vtts vol_opt)
+{
+	reg_base->PADCTLR = (reg_base->PADCTLR & PECI_DVIE) | vol_opt;
+}
+
+static void peci_it8xxx2_rst_status(struct peci_it8xxx2_regs *reg_base)
+{
+	reg_base->HOSTAR = HOSTAR_RST_ANYBIT;
+}
+
+static int peci_it8xxx2_check_host_busy(struct peci_it8xxx2_regs *reg_base)
+{
+	return (reg_base->HOSTAR & HOBY) ? (-EBUSY) : 0;
+}
+
+static int peci_it8xxx2_check_host_finish(const struct device *dev)
+{
+	struct peci_it8xxx2_data *data = dev->data;
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	int ret = k_sem_take(&data->device_sync_sem, K_MSEC(PECI_TIMEOUT_MS));
+
+	if (ret == -EAGAIN) {
+		LOG_ERR("%s: Timeout", __func__);
+		return -ETIMEDOUT;
+	}
+
+	if (peci_regs->HOSTAR != FINISH) {
+		LOG_ERR("[PECI] Error: HOSTAR=0x%02X\r\n", peci_regs->HOSTAR);
+			return -EIO;
+	}
+
+	return 0;
+}
+
+static int peci_it8xxx2_configure(const struct device *dev, uint32_t bitrate)
+{
+	struct peci_it8xxx2_data *data = dev->data;
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	uint8_t hoctl2r_to_write;
+
+	data->bitrate =  bitrate;
+
+	hoctl2r_to_write =
+		(peci_regs->HOCTL2R) & (~(PECI_IT8XXX2_BITRATE_BITS_MASK));
+
+	switch (bitrate) {
+	case PECI_IT8XXX2_BITRATE_2MHZ:
+		break;
+
+	case PECI_IT8XXX2_BITRATE_1MHZ:
+		hoctl2r_to_write |= PECI_IT8XXX2_BITRATE_1MHZ_BITS;
+		break;
+
+	case PECI_IT8XXX2_BITRATE_1P6MHZ:
+		hoctl2r_to_write |= PECI_IT8XXX2_BITRATE_1P6MHZ_BITS;
+		break;
+
+	default:
+		LOG_ERR("[PECI] Error: Specified Bitrate Not Supported\r\n");
+		hoctl2r_to_write |= PECI_IT8XXX2_BITRATE_1MHZ_BITS;
+		data->bitrate =  PECI_IT8XXX2_BITRATE_1MHZ;
+		peci_regs->HOCTL2R = hoctl2r_to_write;
+		return -ENOTSUP;
+	}
+
+	peci_regs->HOCTL2R = hoctl2r_to_write;
+
+	return 0;
+}
+
+static int peci_it8xxx2_enable(const struct device *dev)
+{
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	peci_regs->HOCTLR |= (FIFOCLR|FCSERR_ABT|PECIHEN|CONTROL);
+
+	return 0;
+}
+
+static int peci_it8xxx2_disable(const struct device *dev)
+{
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	peci_regs->HOCTLR &= ~(PECIHEN);
+	return 0;
+}
+
+static void peci_it8xxx2_rst_module(const struct device *dev)
+{
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	peci_regs->HOSTAR = HOSTAR_RST_ANYBIT;
```

Comment:
Replace with `peci_it8xxx2_rst_status(peci_regs);`

Also, should the interrupt status bits be cleared after the global reset at line 234 instead?

Reply:
When it comes to the global reset, I think that the BIOS would reinitialize the PECI module in the POST. However, if the transaction failed to work, it definitely would lead the code to the function `peci_it8xxx2_rst_module ` which resets the PECI module and I believe that the status register has been cleared before module reset. Besides, we also believe that all the PECI regs would be cleared thoroughly when the module is reset by writing the corresponding reset bit to the register RSTC4. 

```
	peci_regs->HOSTAR = HOSTAR_RST_ANYBIT;

	LOG_ERR("[PECI] Module Reset for Status Error.\r\n");
	/* Reset IT8XXX2 PECI Module Thoroughly */
	IT83XX_GCTRL_RSTC4 |= RPECI;
```

Reply:
If setting RPECI clears the pending interrupts, you should delete line 232.

If an explicit clear of the interrupts is needed, pplease change line 232 to call `peci_it8xxx2_rst_status(peci_regs);`

Reply:
Dear Keith, 

So appreciative of your guidance, yes I should delete line 232.
Thanks again.